### PR TITLE
PIMOB-925: Billing form component: summary

### PR DESCRIPTION
**Pull request by ehab-al-cko**
[Jira ticket](https://checkout.atlassian.net/browse/PIMOB-925?atlOrigin=eyJpIjoiMzhkYTM1ZWE5ZDZhNGEzMDljY2U0NmFiZGVjYTZmMGUiLCJwIjoiaiJ9)

Please review this PR firstly https://github.com/checkout/frames-ios/pull/177

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Frames Ios?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [ ] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments
Simulator Screen Shot - iPhone 13 - 2022-06-21 at 14 07 24
Simulator Screen Shot - iPhone 13 - 2022-06-21 at 14 07 26
Simulator Screen Shot - iPhone 13 - 2022-06-21 at 14 07 28

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: Source/UI/Controllers/CardViewController.swift [hunk 1]
```diff
@@ -451,13 +469,55 @@ public class CardViewController: UIViewController,
...
+        updateSummaryValue(with: address.addressLine2)
+        updateSummaryValue(with: address.city)
+        updateSummaryValue(with: address.state)
+        updateSummaryValue(with: address.zip)
+        updateSummaryValue(with: address.country?.name)
+        updateSummaryValue(with: phone.number, withNewLine: false)
+        summaryCellButtonStyle.summary.text = summaryValue
+        cardView.updateBillingFormSummaryView(style: summaryCellButtonStyle)
+
+        func updateSummaryValue(with value: String?, withNewLine: Bool = true) {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/Controllers/CardViewController.swift:492]: updateSummaryValue can be outside the function its a function within a function ?. Also can we add some unit tests for this. Its a valid candidate for unit tests.

File: Source/UI/NewUI/PaymentForm/View/BillingSummary/BillingFormSummaryView.swift [hunk 2]
```diff
@@ -0,0 +1,193 @@
+import UIKit
+
+public protocol BillingFormSummaryViewDelegate: AnyObject {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/NewUI/PaymentForm/View/BillingSummary/BillingFormSummaryView.swift:3]: Are we not using this BillingFormSummaryViewDelegate in sample app ?

File: Source/UI/NewUI/PaymentForm/Default Implementation/BillingFormDetails/Summary/DefaultSummaryCellButtonStyle.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+public struct DefaultSummaryCellButtonStyle: SummaryCellButtonStyle {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/NewUI/PaymentForm/Default Implementation/BillingFormDetails/Summary/DefaultSummaryCellButtonStyle.swift:3]: Can we call this without cell. -> DefaultSummaryButtonStyle ? as merchant can add this ButtonStyle on any button just not a cell button.

File: Source/UI/Views/CardView.swift [hunk 4]
```diff
@@ -128,8 +149,20 @@ public class CardView: UIView {
         }
         stackView.addArrangedSubview(expirationDateInputView)
         stackView.addArrangedSubview(cvvInputView)
+        billingFormEmptyDetailsInputView.removeFromSuperview()
+        billingFormSummaryView.removeFromSuperview()
+        billingDetailsInputView.removeFromSuperview()
         if billingDetailsState != .hidden {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/Views/CardView.swift:155]: can we move line 155 to  165 into a separate function ?

File: iOS Example Frame/iOS Example Frame/ar.lproj/Main.strings [hunk 5]
```diff
@@ -0,0 +1,30 @@
+
+/* Class = "UIButton"; normalTitle = "Custom 1"; ObjectID = "1ce-tx-WHe"; */
```
Comment (deepesh-vasthimal-cko) [inline iOS Example Frame/iOS Example Frame/ar.lproj/Main.strings:2]: Do we need this file to be committed ?

File: Source/Extensions/UIViewExtensions.swift [hunk 6]
```diff
@@ -69,10 +69,28 @@ extension UIView {
...
+            bottomAnchor.constraint(equalTo: view.bottomAnchor),
+        ])
+    }
+}
+
+extension UIView {
+    func disabledAutoresizingIntoConstraints<V: UIView>() -> V? {
+        translatesAutoresizingMaskIntoConstraints = false
+        return self as? V
+    }
```
Comment (harry-brown-cko) [inline Source/Extensions/UIViewExtensions.swift:95]:
```suggestion
    func disabledAutoresizingIntoConstraints() -> Self {
        translatesAutoresizingMaskIntoConstraints = false
        return self
    }
```

**Review by harry-brown-cko — COMMENTED** [review]
one suggestion to make `disabledAutoresizingIntoConstraints` non-optional. should clean up a lot of optional chaining

**Review by harry-brown-cko — APPROVED** [review]

**Pull request merged**